### DISCUSSION TITLE
Get the material property value at 1D boundary

### DISCUSSION THREAD

**Discussion by SomnusYu** [idaholab/moose#27036] [Q&A General]
Hi all,

I would like to get the material property value at the 1D boundary and use this value in ADMaterial. What should I do? Must I use a postprocessor? Is there any other direct way (do not need to modify input file)? 

Thanks!

**Comment by GiudGiud** [comment]
Hello

If you used a postprocessor you would not be able to keep the derivatives with AD. 

Is this material defined all over the domain or only at the boundary?

Guillaume

**Reply by SomnusYu** [reply]
This material defined all over the domain.

**Reply by GiudGiud** [reply]
Are you using distrivuted memory parallelism? (MPI)

**Reply by SomnusYu** [reply]
Does MOOSE use MPI automatically? I do not use MPI.

**Reply by GiudGiud** [reply]
When you are planning to run the simulation, will you use MPI?
I m asking because having derivatives with regards to values on the boundary when the boundary is on another MPI processs s domain is going to be complicated

**Reply by SomnusYu** [reply]
Yes, I think I will use MPI in the future.

**Reply by GiudGiud** [reply]
Ok the postprocessor route won't do AD but it will work fine with MPI.

So maybe you should try that first. Use a postprocessor to compute a single average for boundary material, and use that value everywhere in the domain

**Reply by SomnusYu** [reply]
Thanks!
